src/Console/SelfUpdate/NewVersionChecker.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\SelfUpdate;
+
+use Composer\Semver\Comparator;
+use Composer\Semver\Semver;
+use Composer\Semver\VersionParser;
+
+/**
+ * @internal
+ */
+final class NewVersionChecker
```

Comment:
imo this class does too much things.
it shall:
- provide newest version
- provide newest version for given MAJOR
- expose compare function for 2 provided versions (returning -1, 0, 1)

Reply:
I don't understand your comment: you say it does too much things and then you suggest that it should do more things that it already does (it already provides the newest version and the newest version for current MAJOR).

Reply:
that's the thing - it doesnt. it does **if** they are newer. methods are not atomic. either provide value or compare them, you do both at once.
so if I will want to get newest version, i cannot do it nicely as I need to provide `0.0.0` as base version used for comparision

Reply:
Class updated to have atomic methods.